addons/l10n_ro_edi_etransport/models/l10n_ro_edi_etransport_document.py

```diff
@@ -0,0 +1,563 @@
+
+    l10n_ro_edi_etransport_state = fields.Selection(selection=DOCUMENT_STATES, string="eTransport Status", default='draft', copy=False)
+    l10n_ro_edi_etransport_message = fields.Char(string="Message", copy=False)
+    l10n_ro_edi_etransport_uit = fields.Char(help="UIT of this eTransport document.", copy=False)
+    l10n_ro_edi_etransport_load_id = fields.Char(help="Id of this document used for interacting with the anaf api.", copy=False)
+
+    def _reset_location_type(self):
+        for doc in self:
+            doc.l10n_ro_edi_etransport_start_loc_type_1 = 'location'
+            doc.l10n_ro_edi_etransport_start_loc_type_2 = 'location'
+            doc.l10n_ro_edi_etransport_end_loc_type_1 = 'location'
+            doc.l10n_ro_edi_etransport_end_loc_type_2 = 'location'
+
+    def _reset_location_data(self, location: str):
+        for doc in self:
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_bcp', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_customs_office', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_state_id', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_zip', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_city', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_street', None)
+            setattr(doc, f'l10n_ro_edi_etransport_{location}_other_info', None)
+
+    @api.depends('l10n_ro_edi_etransport_operation_type_id')
+    def _compute_scope_ids(self):
+        for doc in self:
+            scope_domain = []
+
+            if doc.l10n_ro_edi_etransport_operation_type_id:
+                allowed_scope_codes = OPERATION_TYPE_TO_ALLOWED_SCOPE_CODES.get(doc.l10n_ro_edi_etransport_operation_type_id.code, ("9999",))
+                scope_domain = [('code', 'in', allowed_scope_codes)]
+
+                if doc.l10n_ro_edi_etransport_operation_scope_id and doc.l10n_ro_edi_etransport_operation_scope_id.code not in allowed_scope_codes:
+                    doc.l10n_ro_edi_etransport_operation_scope_id = None
+            elif doc.l10n_ro_edi_etransport_operation_scope_id:
+                doc.l10n_ro_edi_etransport_operation_scope_id = None
+
+            doc.l10n_ro_edi_etransport_operation_allowed_scope_ids = doc.env['l10n_ro.edi.etransport.operation.scope'].search(scope_domain)
+
+    @api.depends('picking_id.move_ids')
+    def _compute_l10n_ro_edi_etransport_move_ids(self):
+        for doc in self:
+            doc.l10n_ro_edi_etransport_move_ids = doc.picking_id.move_ids
+
+    @api.depends('picking_id.company_id')
+    def _compute_l10n_ro_edi_etransport_company_id(self):
+        for doc in self:
+            doc.l10n_ro_edi_etransport_company_id = doc.picking_id.company_id
+
+    @api.depends('picking_id.carrier_id')
+    def _compute_l10n_ro_edi_etransport_carrier_id(self):
+        for doc in self:
+            doc.l10n_ro_edi_etransport_carrier_id = doc.picking_id.carrier_id
```

Comment:
yes because the logic to get them differs if the document is part of a batch picking or a picking. what further complexifies your suggestion is that there could also be the case that the batch picking module is not installed, then we would have to call `getattr` to see if the `batch_id` field exists. using the compute functions is IMO way more elegant

Reply:
I see, I haven't checked the batch module yet. Ok then